llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -1430,6 +1430,13 @@ private SelectHostResult selectHost(TaskInfo request, Map<String, List<NodeInfo>
         boolean requestedHostsWillBecomeAvailable = false;
         for (String host : requestedHosts) {
           prefHostCount++;
+
+          // Check if the host is removed from the registry after availableHostMap is created.
+          Set<LlapServiceInstance> activeInstancesByHost = activeInstances.getByHost(host);
+          if (activeInstancesByHost == null || activeInstancesByHost.isEmpty()) {
+            continue;
+          }
+
           // Pick the first host always. Weak attempt at cache affinity.
           if (availableHostMap.containsKey(host)) {
```

Comment:
I would avoid the continue statement above and modify the condition to:
if (availableHostMap.containsKey(host) && activeInstancesByHost != null && !activeInstancesByHost.isEmpty())

Reply:
i think having this separate check makes the code more readable.